multiraft/multiraft.go

```diff
@@ -33,6 +33,8 @@ import (
 
 const (
 	noGroup = proto.RaftID(0)
+
+	reqBufferSize = 100
```

Comment:
A list+condvar is essentially a channel with unlimited size, which is kind of scary. Also, we need to be able to `select` on this channel so if we used a list+condvar we'd need a separate goroutine to copy messages from the list onto an unbuffered channel as they become available.

Blocking here is not the end of the world; it just results in backpressure on the TCP connection, so it might be reasonable to just let it block when the channel fills up. The biggest risk of blocking in the rpc thread is deadlock, if we can't make progress due to something that indirectly depends on some future RPC getting through to this node. But I don't *think* that can happen with raft messages; we should always be able to clear any backlog at the raft level without any non-local resources. 

I think that as long as we only block the rpc thread in exceptional circumstances then the bounded channel is probably the best solution. If this does turn out to be a problem then I think I'd rather start dropping messages than move to buffering them in an unbounded list.